github_utils.py

```diff
@@ -578,3 +578,70 @@ def mark_commit_status(commit, repository, context="default", state="pending", u
         params['context'] = s['context']
         github_api('/repos/%s/statuses/%s' % (repository, commit), params=params)
   return
+
+def get_branch(repository, branch_name):
+    get_gh_token(repository)
+    data = github_api("/repos/%s/branches/%s" % (repository, branch_name), method="GET")
+    return data
+
+
+def get_git_tag(repository, tag_name):
+    get_gh_token(repository)
+    data = github_api(
+        "/repos/%s/git/ref/tags/%s" % (repository, tag_name), method="GET"
+    )
+    return data
+
+
+def create_git_tag(
+    repository, tag_name, commit_sha, tagger_name, tagger_email, tag_date=None
+):
+    get_gh_token(repository)
+    if tag_date is None:
+        tag_date = datetime.datetime.now()
+    tag_date_s = tag_date.replace(microsecond=0).isoformat()
+    params = {
+        "tag": tag_name,
+        "message": "Create tag %s" % tag_name,
+        "object": commit_sha,
+        "type": "commit",
+        "tagger": {"name": tagger_name, "email": tagger_email, "date": tag_date_s},
+    }
+    data = github_api("/repos/%s/git/tags" % repository, method="POST", params=params)
+    tag_sha = data["sha"]
+    params = {"ref": "refs/tags/%s" % tag_name, "sha": tag_sha}
+    github_api("/repos/%s/git/refs" % repository, method="POST", params=params)
+
+
+def get_commit_tags(repository, commit_sha, all_tags=False):
+    get_gh_token(repository)
+    res = []
+    data = github_api("/repos/%s/tags" % repository, all_pages=all_tags, method="GET")
+    for tag in data:
+        if tag["commit"]["sha"] == commit_sha:
+            res.append(tag["name"])
+
+    return res
+
+
+def get_commits(repository, branch, until, per_page=1):
```

Comment:
@iarspider , did you check how many commints it returns ? May be we should also set `since` param

Reply:
I explicitly ask for only 1 commit (`per_page=1`)

Reply:
but this might not be the correct commit. For example for 11h00 IB, one could have merged a PR after 11h00 which contains a commit made at 10h58. This should return the commit matched by 
```
NEW_HASH=`git rev-list -n 1 --before="$DAY $HOUR:00" --first-parent $RELEASE_BRANCH`
```
right? So it should be the last merge commit on `$RELEASE_BRANCH` branch before the selected hour

Reply:
So I need to query 100 commits, and iterate until I find a merge commit, right?

Reply:
(100 or whatever max number of commits fit on one page)